woudc_data_registry/epicentre/instrument.py

```diff
@@ -188,13 +187,26 @@ def update(ctx, identifier, station, dataset,
     """Update instrument information"""
 
     instrument_ = {}
+    registry_ = registry.Registry()
 
     if station:
         instrument_['station_id'] = station
     if dataset:
         instrument_['dataset_id'] = dataset
-    if station and contributor:
-        instrument_['deployment_id'] = ':'.join([station, contributor])
+    if station or contributor:
+        temp_index = registry_.query_full_index(Instrument)
+        i = 0
+        for i in range(len(temp_index)):
+            if temp_index[i].instrument_id == identifier:
+                break
```

Comment:
Even though I suggested not to use `registry_` directly anyways, want to leave a note here:

What we're doing here is grabbing the *entire* instruments table from the database and doing a linear search. Since instruments is one of the larger tables, that could be quite expensive. We already have a mandatory identifier and so know what ID to look for, so some kind of query-by-ID would be much more efficient. You can look for examples of other parts of the project using `registry_.query_by_field` to get that effect, e.g. https://github.com/woudc/woudc-data-registry/blob/master/woudc_data_registry/processing.py#L840.

Try to avoid doing stuff in Python (like searching) wherever possible, that's what the database is there for. Again, my suggestion is to use `metadata.get_metadata` instead of directly using the registry at all.